contracts/pool/base/PoolVestingPeriod.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "forge-std/console.sol";
+import {EnumerableMap} from "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+import {PoolBase} from "./PoolBase.sol";
+
+/// @title Base contract for Pool with Vesting period
+/// @author Tigran Arakelyan
+/// @dev Should be inherited
+abstract contract PoolVestingPeriod is PoolBase {
+    using EnumerableMap for EnumerableMap.AddressToUintMap;
+
+    EnumerableMap.AddressToUintMap private holdersToEffectiveDepositDate;
+
+    constructor(address _asset, string memory _tokenName, string memory _tokenSymbol)
+    PoolBase(_asset, _tokenName, _tokenSymbol) {}
+
+    /*
+    ERC20 overrides
+    */
+
+    function transfer(address to, uint amount) public override returns (bool) {
+        _updateHolder(msg.sender, to, amount);
+        return super.transfer(to, amount);
+    }
+
+    function transferFrom(address from, address to, uint amount) public override returns (bool) {
+        _updateHolder(msg.sender, to, amount);
+        return super.transferFrom(from, to, amount);
+    }
+
+    /*
+    Helpers
+    */
+
+    /// @notice Get the count of holders
+    function getHoldersCount() public view returns (uint256) {
+        return holdersToEffectiveDepositDate.length();
+    }
+
+    /// @notice Get holders
+    function getHolders() public view returns (address[] memory) {
```

Comment:
It's very expensive function to be called on-chain, so I guess it was never meant to be called so (at least no such calls were found within protocol). That said I would consider making it external. 

```suggestion
    function getHolders() external view returns (address[] memory) {
```